scripts/cmake/xtensa-toolchain.cmake

```diff
@@ -56,6 +56,7 @@ endif()
 
 find_program(CMAKE_LD NAMES "${CROSS_COMPILE}ld" PATHS ENV PATH NO_DEFAULT_PATH)
```

Comment:
@ktrzcinx I'm totally puzzled by this PR.

Please explain in the commit message what is the ABI change that we are fixing. Also explain WHY this change is needed.

Reply:
changed

Reply:
I think a few words in the code itself wouldn't hurt either, for now it's just pure maths :-)